src/components/__tests__/QuestionnaireItem.cy.js

```diff
@@ -0,0 +1,153 @@
+    });
+  });
+
+  it('has translation for all strings', () => {
+    cy.testLanguageStringsInContext(
+      ['buttonFillQuestionnaire'],
+      'profile',
+      i18n,
+    );
+  });
+
+  context('desktop - target _blank', () => {
+    beforeEach(() => {
+      cy.wrap(questionnaires[0]).as('questionnaire');
+      cy.get('@questionnaire').then((questionnaire) => {
+        cy.mount(QuestionnaireItem, {
+          props: {
+            questionnaire: questionnaire,
+          },
+        });
+      });
+      cy.viewport('macbook-16');
+    });
+
+    coreTests();
+    iconTests();
+  });
+
+  context('desktop - target _self', () => {
+    beforeEach(() => {
+      cy.wrap(questionnaires[1]).as('questionnaire');
+      cy.get('@questionnaire').then((questionnaire) => {
+        cy.mount(QuestionnaireItem, {
+          props: {
+            questionnaire: questionnaire,
+          },
+        });
+      });
+      cy.viewport('macbook-16');
+    });
+
+    coreTests();
+
+    it('does not render an external link icon in the button', () => {
+      cy.dataCy(questionnaireButtonIcon).should('not.exist');
+    });
+  });
+
+  context('mobile', () => {
+    beforeEach(() => {
+      cy.wrap(questionnaires[0]).as('questionnaire');
+      cy.get('@questionnaire').then((questionnaire) => {
+        cy.mount(QuestionnaireItem, {
+          props: {
+            questionnaire: questionnaire,
+          },
+        });
+      });
+      cy.viewport('iphone-6');
+    });
+
+    coreTests();
+    iconTests();
+  });
+
+  function coreTests() {
+    it('renders component', () => {
+      cy.dataCy(questionnaireItem).should('be.visible');
+    });
+
+    it('displays the questionnaire title', () => {
+      cy.get('@questionnaire').then((questionnaire) => {
+        cy.dataCy(questionnaireTitle)
+          .should('have.css', 'font-size', '16px')
+          .and('have.css', 'font-weight', '400')
+          .and('contain', questionnaire.title);
+      });
+    });
+
+    it('renders the questionnaire image', () => {
+      cy.get('@questionnaire').then((questionnaire) => {
+        cy.dataCy(questionnaireAvatar)
+          .should('be.visible')
+          .invoke('height')
+          .should('be.eq', avatarSize);
+        cy.dataCy(questionnaireAvatar)
+          .invoke('width')
+          .should('be.eq', avatarSize);
+        cy.dataCy(questionnaireImage)
+          .find('img')
+          .should('have.attr', 'src', questionnaire.image.src)
+          .and('have.attr', 'alt', questionnaire.image.alt);
+      });
+    });
+
+    it('renders a button with correct attributes and text', () => {
+      cy.get('@questionnaire').then((questionnaire) => {
+        cy.dataCy(questionnaireButton)
+          .should('be.visible')
+          .and('have.attr', 'href', questionnaire.link.url)
+          .and('have.attr', 'target', questionnaire.link.target)
+          .and('contain', i18n.global.t('profile.buttonFillQuestionnaire'));
+      });
+    });
+
```

Comment:
You can extend tests by test URL (request/response)

Existed code example:

https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/47bc98c6cb137b765dfa2d83c7790dbd8bf9c222/src/components/__tests__/FooterBar.cy.js#L131-L140

Reply:
I added a valid URL to fixture + the test.